### PR TITLE
Remove SetConsumerError from sink test helpers

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes https://github.com/open-telemetry/opentelemetry-collector/issues/2477

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: receiver/opencensusreceiver/opencensus_test.go [hunk 1]
```diff
@@ -612,3 +614,41 @@ func TestOCReceiverMetrics_HandleNextConsumerResponse(t *testing.T) {
 		}
 	}
 }
+
+type errOrSinkConsumer struct {
```
Comment (bogdandrutu) [inline receiver/opencensusreceiver/opencensus_test.go:618]: In the future if this gets copied consider to provide an util.

Comment (tigrannajaryan) [reply receiver/opencensusreceiver/opencensus_test.go:618]: Why create duplicates in receivers now? Maybe move to `internal` if you don't want to have this public.

Comment (bogdandrutu) [reply receiver/opencensusreceiver/opencensus_test.go:618]: That is a good option, will do that. Indeed I am not sure about that code/api so prefer to have it non-public

Comment (bogdandrutu) [reply receiver/opencensusreceiver/opencensus_test.go]: Done.

**Review by tigrannajaryan — APPROVED** [review]
LGTM

**Pull request merged**